sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/util/_configurations.py

```diff
@@ -109,6 +95,52 @@ def _default_sampling_ratio(configurations):
     configurations[SAMPLING_RATIO_ARG] = default
 
 
-# TODO: Placeholder for future configuration
-def _default_disable_azure_core_tracing(configurations):
-    configurations[DISABLE_AZURE_CORE_TRACING_ARG] = False
+def _merge_nested_dicts_in_place(d1, d2):
+    for key, value in d2.items():
+        if key in d1 and isinstance(d1[key], dict) and isinstance(value, dict):
+            _merge_nested_dicts_in_place(d1[key], value)
+        else:
+            d1[key] = value
+
+
+def _merge_nested_dicts(d1, d2):
+    merged = d1.copy()
+    for key, value in d2.items():
+        if key in merged and isinstance(merged[key], dict) and isinstance(value, dict):
+            _merge_nested_dicts_in_place(merged[key], value)
+        else:
+            merged[key] = value
+    return merged
+
+
+def _default_instrumentation_options(configurations):
+    disabled_instrumentation = environ.get(
+        OTEL_PYTHON_DISABLED_INSTRUMENTATIONS, ""
+    )
+    disabled_instrumentation = disabled_instrumentation.split(",")
+    # to handle users entering "requests , flask" or "requests, flask" with spaces
+    disabled_instrumentation = [
+        x.strip() for x in disabled_instrumentation
+    ]
+
+    default_instrumentation_options = {}
+    for lib_name in _FULLY_SUPPORTED_INSTRUMENTED_LIBRARIES:
+        disabled_by_env_var = lib_name in disabled_instrumentation
+        default_instrumentation_options[lib_name] = {"enabled": not disabled_by_env_var}
+    for lib_name in _PREVIEW_INSTRUMENTED_LIBRARIES:
+        default_instrumentation_options[lib_name] = {"enabled": False}
+
+    # Explicit configuration takes priority over environment variables
+    instrumentation_options = configurations.get(INSTRUMENTATION_OPTIONS_ARG, {})
+    merged_instrumentation_options = _merge_nested_dicts(default_instrumentation_options, instrumentation_options)
```

Comment:
As discussed, i think we just want to iterate through the names and apply the merge, instead of merging a nested dict.

Reply:
For the record, stashing the nested merge as "instrumentation-config" if we want to enable configuration.

Reply:
Done.